resources/ner_text_extraction_pipeline/pipeline.py

```diff
@@ -0,0 +1,237 @@
+import argparse
+import csv
+import json
+import os
+import re
+import requests
+import torch
+
+from multiprocessing.pool import Pool
+from transformers import BertTokenizerFast, BertForTokenClassification
+
+
+MAX_NER_INPUT_LEN = 128
+ROR_URL = "https://api.ror.org/organizations"
+
+
+def load_model(model_path):
+    """
+    Load NER model.
+    Copied from https://github.com/ror-community/affiliation-matching-experimental/tree/main/ner_tests/inference
+
+    :param model_path: Path of the model directory
+    :return: Model object
+    """
+
+    model = BertForTokenClassification.from_pretrained(model_path)
+    model = model.cpu()
+    return model
+
+
+def preprocess_input(tokenizer, ner_input):
+    """
+    Preprocess input for NER model.
+    Copied from https://github.com/ror-community/affiliation-matching-experimental/tree/main/ner_tests/inference
+
+    :param tokenizer: NER tokenizer object
+    :param ner_input: A list of input tokens
+    :return: Preprocessed input object
+    """
+
+    return tokenizer(
+        ner_input,
+        is_split_into_words=True,
+        return_offsets_mapping=True,
+        padding="max_length",
+        truncation=True,
+        max_length=MAX_NER_INPUT_LEN,
+        return_tensors="pt",
+    )
+
+
+def ner_inference(model, preprocessed_input):
+    """
+    Perform the NER inference.
+    Copied from https://github.com/ror-community/affiliation-matching-experimental/tree/main/ner_tests/inference
+
+    :param model: NER model object
+    :param preprocessed_input: Preprocessed input object
+    :return: NER token predictions
+    """
+
+    ids = preprocessed_input["input_ids"]
+    mask = preprocessed_input["attention_mask"]
+    outputs = model(ids, attention_mask=mask)
+    logits = outputs[0]
+    active_logits = logits.view(-1, model.num_labels)
+    predictions = torch.argmax(active_logits, axis=1)
+    return predictions
+
+
+def postprocess_output(tokenizer, ner_input, token_predictions):
```

Comment:
`tokenizer` is not used in this function

Reply:
Removed.